lightning/src/routing/router.rs

```diff
@@ -1551,7 +1666,7 @@ where L::Target: Logger {
 		// If a caller provided us with last hops, add them to routing targets. Since this happens
 		// earlier than general path finding, they will be somewhat prioritized, although currently
 		// it matters only if the fees are exactly the same.
-		for route in payment_params.route_hints.iter().filter(|route| !route.0.is_empty()) {
+		for route in route_hints_ref.iter().filter(|route| !route.0.is_empty()) {
```

Comment:
I pushed a fixup for this. It's a bit more code. Not as much reduction as one might expect because you run into `borrowed value does not live long enough` if you don't marshall the blinded route hints vec near the top of the method

Reply:
Hmm.. if we keep the fixup, we can have a `Vec<RouteHintHop>` instead of a `Vec<Vec<RouteHintHop>>`

Reply:
So I'm gonna revert to the previous version here, because I realized that we'd also need to also duplicate the logic that adds entries for first hop<>route hint direct channel candidates (~30 lines, for calculating the blinded path fees, penalty, etc).